doc/support.rst

```diff
@@ -1,10 +1,137 @@
 .. _support:
 
+=======================
 Support and Development
------------------------
+=======================
 
 To get help using the **signac** package, either send an email to `signac-support@umich.edu <mailto:signac-support@umich.edu>`_ or join the `signac gitter chatroom <https://gitter.im/signac/Lobby>`_.
 
 The **signac** package is hosted on `GitHub <https://github.com/glotzerlab/signac.git>`_ and licensed under the open-source BSD 3-Clause license.
 Please use the `repository's issue tracker <https://github.com/glotzerlab/signac/issues>`_ to report bugs or request new features.
 
+
+Code contributions
+==================
+
+This project is open-source.
+Users are highly encouraged to contribute directly by implementing new features and fixing issues.
+Development for packages as part of the **signac** framework should follow the general development guidelines outlined `here <http://docs.signac.io/en/latest/community.html#contributions>`_.
+
+A brief summary of contributing guidelines are outlined in the `CONTRIBUTING.md <https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md>`_ file as part of the repository.
+All contributors must agree to the `Contributor Agreement <https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md>`_ before their pull request can be merged.
+
+Set up a development environment
+--------------------------------
+
+Start by `forking <https://github.com/glotzerlab/signac/fork/>`_ the project.
+
+We highly recommend to setup a dedicated development environment,
+for example with `venv <https://docs.python.org/3/library/venv.html>`_:
+
+.. code-block:: bash
+
+    ~ $ python -m venv ~/envs/signac-dev
+    ~ $ source ~/envs/signac-dev/bin/activate
+    (signac-dev) ~ $ pip install six flake8
+
+or alternatively with `conda <https://conda.io/docs/>`_:
+
+.. code-block:: bash
+
+    ~ $ conda create -n signac-dev python=3 six flake8
+    ~ $ activate signac-dev
+
+Then clone your fork and install the package from source with:
+
+.. code-block:: bash
+
+    (signac-dev) ~ $ cd path/to/my/fork/of/signac
+    (signac-dev) signac $ pip install -e . -r requirements-dev.txt
+
+The ``-e`` option stands for *editable*, which means that the package is directly loaded from the source code repository.
+That means any changes made to the source code are immediately reflected upon reloading the Python interpreter.
+
```

Comment:
I would say that keeping the fork synchronized is the responsibility of the developer. A non-synchronized fork would lead to messy pull requests and that's when we would definitely notice it. Otherwise we wouldn't really care at all.